File: Unity/TSProj/src/mods/ModTheSpire/Scripts/Events/HpBlockBrokenEffect.ts
import { FairyGUI } from "csharp";
import { AbstractEffect } from "../Effect/AbstractEffect";
import { View_HpBlockBrokenEffect } from "../Gen/View/ModTheSpire_Effect";

export class HpBlockBrokenEffect extends AbstractEffect
{
    public Com:FairyGUI.GComponent;
    public constructor(private _X:number,private _Y:number) {
        super();
        this.Com = <FairyGUI.GComponent>View_HpBlockBrokenEffect.CreateInstance();
        this.Com.SetPosition(_X,_Y,0);    
    }
}

File: Unity/Assets/Scripts/ModTheSpire/Effect/Combat/HpBlockBrokenEffect.cs
using ModTheSpire.ModTheSpire_Effect;

namespace ModTheSpire.Effect.Combat
{
    public class HpBlockBrokenEffect : AbstractEffect
    {
        private View_HpBlockBrokenEffect mCom;
        public HpBlockBrokenEffect(float x, float y)
        {
            mCom = View_HpBlockBrokenEffect.CreateInstance();
            mCom.SetPosition(x,y,0);
            mCom.t0.Play(1,0,Finish);
        }

        public override void Dispose()
        {
            base.Dispose();
            mCom.Dispose();
        }
    }
}